markdown/view.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react'
+import Markdown from 'react-markdown'
+
+interface Props {
+  markdown: string;
+  onSave: (newMarkdown: string) => Promise<void>;
+}
+
+export const View: React.FC<Props> = (props) => {
+  const [phase, setPhase] = React.useState<'saving' | 'rendering' | 'editing'>('rendering')
+  const [rawText, setRawText] = React.useState(props.markdown)
+
+  function storeMarkdown () {
+    setPhase('saving')
+    props.onSave(rawText).then(() => {
+      setPhase('rendering')
+    })
+  }
+
+  if (phase === 'saving') {
```

Comment:
I like this approach better than the conditionals in my experiment - I think this is easier to read.